src/svd/two-sided/svd.cpp

```diff
@@ -0,0 +1,132 @@
+#include "svd.hpp"
+#include <assert.h>
+#include <math.h>
+#include <algorithm>
+#include "matrix.hpp"
+#include "nsvd.hpp"
+#include "types.hpp"
+#include "util.hpp"
+
+void matrix_frobenius(matrix_t m, double* norm, double* off_norm);
```

Comment:
If these functions are only used in this `cpp` file, we can consider making them static.

Reply:
Good idea!